### PR TITLE
Add tests for zstd content-encoding

### PR DESCRIPTION
**Pull request by nidhijaju**
This PR adds WPTs for decompressing different types of zstd-encoded content: simple, big, invalid, and an incompatibly large window size. This also reorganizes the tests so they are separated by compression scheme.

### CODE REVIEW

**Review by ricea — APPROVED** [review]
lgtm, nice.

**Comment by nidhijaju** [comment]
@annevk Could you please take a look to see if these WPTs look reasonable? Thank you!

**Review by annevk — COMMENTED** [review]
This looks like reasonable zstd coverage. Although perhaps there should be more payloads? Does this also test a variety of invalid payloads and ensure the error handling is correct?

The other thing that is ideally tested is Content-Encoding handling. When it contains multiple values, strange whitespace characters, etc. That can be separate from this PR however.

File: fetch/content-encoding/zstd/big-window-zstd-body.https.any.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// META: global=window,worker
+
+promise_test(async t => {
+  const response = await fetch('resources/big.window.zst');
+  assert_true(response.ok);
+  await promise_rejects_js(t, TypeError, response.text());
+}, 'Consuming the body of a resource with too large of a zstd window size should reject');
```
[the remark this answers is not in the document]
Comment (ricea) [reply fetch/content-encoding/zstd/big-window-zstd-body.https.any.js:7]: RFC 8878 has some fuzzy language regarding which window sizes should be accepted. https://github.com/facebook/zstd/issues/2713 is a discussion of how to make this a concrete requirement. Although we don't have consensus yet on how the requirement should be specified, the value of an 8 MB window size seems to be mostly uncontroversial.

[the remark this answers is not in the document]
Comment (nidhijaju) [reply fetch/content-encoding/zstd/big-window-zstd-body.https.any.js]: Done.

**Comment by nidhijaju** [comment]
> Does this also test a variety of invalid payloads and ensure the error handling is correct?

I added some tests for invalid payloads similar to the ones for gzip. Please take a look.

**Review by annevk — APPROVED** [review]
Thanks!

I think at some point we might want more exhaustive zstd coverage. Or is there another test suite elsewhere that implementations use?

I also still think we need more Content-Encoding coverage, but it doesn't have to all be in this PR.

**Comment by nidhijaju** [comment]
Thank you for the review!

> is there another test suite elsewhere that implementations use?

We have a set of [unit tests](https://source.chromium.org/chromium/chromium/src/+/main:net/filter/zstd_source_stream_unittest.cc) in Chromium that we use to test the zstd decompression. Some of them can potentially be converted to WPTs to increase coverage in the future.

**Pull request merged**